irods-vfs-impl/src/main/java/org/irods/nfsrods/vfs/IRODSVirtualFileSystem.java

```diff
@@ -1645,15 +1663,101 @@ private static Inode toFh(long _inodeNumber)
         return Inode.forFile(Longs.toByteArray(_inodeNumber));
     }
 
+    private String tryAndFindPathFromGenQuery(long _inodeNumber, String option) {
+        // if getPath() returns null, we'll use GenQuery to find the path
+        log_.debug("""
+                vfs::tryAndFindPathFromGenQuery
+                tryAndFindPathFromGenQuery - _inodeNumber = {}
+                tryAndFindPathFromGenQuery - option       = {}""", 
+                _inodeNumber, option);
+
+        String strPath = null;
+        IRODSGenQueryBuilder builder;
+        IRODSGenQueryExecutor gqe;
+        IRODSGenQueryFromBuilder query;
+        IRODSQueryResultSet resultSet;
+
+        if (option.equals("coll")) {
```

Comment:
Will change to enumeration, just wanted to spin up something quick with the string. 

Just changed to use the `ObjectType` enum. Since we're doing this, if `_option` is `null`, the `else` case will be hit and an error message will be printed